### DISCUSSION TITLE
Navigation with browser buttons works unexpectedly

### DISCUSSION THREAD

**Discussion by lizmikhailava** [vuejs/router#2225] [Help and Questions]
I'm using web history router and navigate to particular route using push(), and when I try to navigate using browser buttons (back and forward), it doesn't work properly. Maybe it happens because I'm navigating between the same url but with different queries. Could you tell me, whether I'm using router wrong or it's a bug?

**Comment by posva** [comment]
you will need to share some code. You can even put it in a [playground](https://play.vuejs.org/#eNq1U8lu2zAQ/RVCLSAHsMVmQQ6CkqZNgy7oEqRFL1UPjEXbTCSS4OK4MPTvHZEUZceJgR56MExxZt68efO4ThrCeHankzxhjRTKoDWaKkoMfSMlatFMiQalS0vTkocEJayhKkQy7D+HcFfXx4iUma8teUQdwe+g5AhlVtORL/ffjbDcjNIXUJYeJOMklAO3wtBG1lB/3iUW0v3BQRsl+Pz80ipFeaCGJDGLvMAhhtZr9NIFspmt62sIorZ1MNjjFLbu8WoWTnC+ccw+M36PciPOymSNOGlojtKK6MWtIKpKUVsm5+8FMgLheFvgobQHxhH5n3uMYSBFGp3DaliVo8OO/m5XfPif+x490/doX98CB3GLzmh91Of/ZPQB4ZAWwgWOmwYDeHM8ac8vtBHqzwemDfyNw6XH3bTtJNozGvRdz9z17626dTuYVnAdbKXRGfrVcY1SLURDnUpgN5RiOE4FtODgxPxRl3a8Vbktsi8fBM1Z9XovWMl/PyKngNymBKN112/h1cmfkmx04Cj50UrewgssOV05hSo6I7buoWEPO+LsPsmKLcN2o2viu/M38dX51O1N+91MGiJh24IDvpugDAFdJgDo8ctk2Gt3XSYLY6TOMbZc3s8z0A0PGRcn2XH2Cnpqs3GbUd1MbpV40M5eZeLE6LAvIAlXdGmEqPWESPZci53Ei9PsNDsE499iQMeMV3TlsDtoGL2FMY2Gpc3Y/NGQ3apZTdU3aRgsdWtYUtfi4ZO7M8rSSHS6oNP7J+7v9MpTvlYUplvSjeEMUXNqfPjq+1e6gnMMNqKyNWTvCd5QLWrbcfRpby2vgPZGnmP70e2M8fkPfbUylOt+qI6oU8Plu0Ve7hl9oHucnUQV27/hwj7r)
